### PR TITLE
Python install requires

### PR DESCRIPTION
**Pull request by cjermain**
This PR ties the `requirements.txt` file directly to `install_requires` in `setup.py`. This is aimed at making it easier to keep the versions up to date, and to avoid either place getting stale. This PR also removes `six` as a requirement, since it does not appear to be used in the package. `protobuf` becomes a required package in `install_requires` based on this change.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -201,6 +201,9 @@ def run(self):
     cmd_classes['bdist_wheel'] = bdist_wheel
 cmd_classes['build_ext'] = build_ext
 
+with open('requirements.txt') as f:
```
Comment (snnn) [inline setup.py:204]: You may revert the changes in this file first. The others looks good.

Comment (cjermain) [reply setup.py]: It looks like there is already some logic for finding the `README.rst`, so I've copied that pattern to find the `requirements.txt`. I can revert it if it doesn't resolve the path correctly.
https://github.com/microsoft/onnxruntime/blob/5aab2671f8346a1489ac8ebf4b364544c6707062/setup.py#L176-L183

**Comment by cjermain** [comment]
Thanks @snnn!
